test/unit/sql/generateJoin.test.js

```diff
@@ -0,0 +1,308 @@
+  , Sequelize = require(__dirname + '/../../../lib/sequelize')
+  , util      = require('util')
+  , _         = require('lodash')
+  , expectsql = Support.expectsql
+  , current   = Support.sequelize
+  , sql       = current.dialect.QueryGenerator;
+
+// Notice: [] will be replaced by dialect specific tick/quote character when there is not dialect specific expectation but only a default expectation
+
+suite(Support.getTestDialectTeaser('SQL'), function() {
+  suite('generateJoin', function () {
+    var testsql = function (path, options, expectation) {
+
+      let name = `${path}, ${util.inspect(options, { depth: 10 })}`;
+      
+      Sequelize.Model.$conformOptions(options);
+      options = Sequelize.Model.$validateIncludedElements(options);   
+
+      let include = _.at(options, path)[0];
+
+      test(name, function () {
+
+        let join = sql.generateJoin(include,
+          {
+            options,
+            subQuery: options.subQuery === undefined ? options.limit && options.hasMultiAssociation : options.subQuery
+          }
+        );
+
+        return expectsql(`${join.join} ${join.body} ON ${join.condition}`, expectation);
+      });
+    };
+
+    var User = current.define('User', {
+      id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true,
+        field: 'id_user'
+      },
+      companyId: {
+        type: DataTypes.INTEGER,
+        field: 'company_id'
+      }
+    }, {
+      tableName: 'user'
+    });
+    var Task = current.define('Task', {
+      title: Sequelize.STRING,
+      userId: {
+        type: DataTypes.INTEGER,
+        field: 'user_id'
+      }
+    }, {
+      tableName: 'task'
+    });
+
+    var Company = current.define('Company', {
+      name: Sequelize.STRING,
+      ownerId: {
+        type: Sequelize.INTEGER,
+        field: 'owner_id'
+      },
+      public: {
+        type: Sequelize.BOOLEAN
+      }
+    }, {
+      tableName: 'company'
+    });
+
+    var Profession = current.define('Profession', {
+      name: Sequelize.STRING
+    }, {
+      tableName: 'profession'
+    });
+
+    User.Tasks = User.hasMany(Task, {as: 'Tasks', foreignKey: 'userId'});
+    User.Company = User.belongsTo(Company, {foreignKey: 'companyId'});
+    User.Profession = User.belongsTo(Profession, {foreignKey: 'professionId'});
+    Profession.Professionals = Profession.hasMany(User, {as: 'Professionals', foreignKey: 'professionId'});
+    Company.Employees = Company.hasMany(User, {as: 'Employees', foreignKey: 'companyId'});
+    Company.Owner = Company.belongsTo(User, {as: 'Owner', foreignKey: 'ownerId'});
+
+    /*
+     * BelongsTo
+     */
+
+    testsql(
+      "include[0]",
+      {
+        model: User,
+        include: [
+          User.Company
+        ]
+      },
+      {
+        default: "LEFT OUTER JOIN [company] AS [Company] ON [User].[company_id] = [Company].[id]"
+      }
+    );
+
+
+    testsql(
+      "include[0]",
```

Comment:
before the test description has some info on the parameters, maybe do something like `'include[0](where, or)`